lib/std/event/lock.zig

```diff
@@ -16,107 +16,90 @@ const Loop = std.event.Loop;
 /// Allows only one actor to hold the lock.
 /// TODO: make this API also work in blocking I/O mode.
 pub const Lock = struct {
-    shared: bool,
-    queue: Queue,
-    queue_empty: bool,
+    mutex: std.Mutex = std.Mutex{},
+    head: usize = UNLOCKED,
 
-    const Queue = std.atomic.Queue(anyframe);
+    const UNLOCKED = 0;
+    const LOCKED = 69;
```

Comment:
what conviction do we have that this isn't a valid value for `head`? (hmmm; can we ensure alignment of `Waiter` is > 1 so that any odd number would work?) in any case we need a comment

Reply:
It was chosen pretty much arbitrarily on the grounds of the waiter nodes having a higher alignment. Will add more docs to clarify .

Reply:
It's a safe assumption that at the very least the first page of memory is unmapped. On freestanding where that may not be the case, this Lock implementation would be invalid anyway since it depends on OS syscalls for the mutex.